### PR TITLE
[GEOS-10680] GeoFence: refactoring

### PR DESCRIPTION
**Pull request by etj**
[GEOS-10680](https://osgeo-org.atlassian.net/browse/GEOS-10680)

- Many classes belonging logically in `geofence-server` were located in `geofence`, and have been moved (more info in the GEOS ticket)
- All classes in `geofence-server` have been moved into `org.geoserver.geofence.server` hierarchy.

This PR is mainly about moving files (and replacing java packages).  
There is a single new file, which is created by the splitting of the XStream init'ter for classes in `geofence` and `geofence-server`; also, one more Spring bean is created for it.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/extension/geofence-server/src/main/java/org/geoserver/geofence/server/xstream/GeoFenceServerXStreamInitializer.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/* (c) 2015 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2015 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
```
Comment (aaime) [inline src/extension/geofence-server/src/main/java/org/geoserver/geofence/server/xstream/GeoFenceServerXStreamInitializer.java:5]:
```suggestion
/* (c) 2022 Open Source Geospatial Foundation - all rights reserved
 * This code is licensed under the GPL 2.0 license, available at the root
 * application directory.
 */
```

**Review by aaime — APPROVED** [review]
Looks good, only a minor issue in the copyright header of a new file (provided suggestion)

**Comment by etj** [comment]
Before merging this PR, I'd wait for the commit for [GEOS-10682](https://osgeo-org.atlassian.net/browse/GEOS-10682) to be ready, bc it's bringing some more refactoring e.g. a `geofence/` own subdir, since the submodules are increasing in number (will also be handy for https://osgeo-org.atlassian.net/browse/GEOS-9904)

**Comment by aaime** [comment]
Closing as superceded by #6227, which contains an updated version of the commit for GEOS-10680

**Pull request closed without merging**